src/services/textChanges.ts

```diff
@@ -446,38 +525,51 @@ namespace ts.textChanges {
         }
 
         private computeNewText(change: Change, sourceFile: SourceFile): string {
-            if (!change.node) {
+            if (change.kind === ChangeKind.Remove) {
                 // deletion case
                 return "";
             }
+
             const options = change.options || {};
-            const nonFormattedText = getNonformattedText(change.node, sourceFile, this.newLine);
+            let text: string;
+            const pos = change.range.pos;
+            const posStartsLine = getLineStartPositionForPosition(pos, sourceFile) === pos;
+            if (change.kind === ChangeKind.ReplaceWithMultipleNodes) {
+                const parts = change.nodes.map(n => this.getFormattedTextOfNode(n, sourceFile, pos, options));
+                text = parts.join(change.options.nodeSeparator);
+            }
+            else {
+                text = this.getFormattedTextOfNode(change.node, sourceFile, pos, options);
+            }
+            // strip initial indentation (spaces or tabs) if text will be inserted in the middle of the line
+            // however keep indentation if it is was forced
+            text = (posStartsLine || options.indentation !== undefined) ? text : text.replace(/^\s+/, "");
+            return (options.prefix || "") + text + (options.suffix || "");
+        }
+
+        private getFormattedTextOfNode(node: Node, sourceFile: SourceFile, pos: number, options: ChangeNodeOptions): string {
+            const nonformattedText = getNonformattedText(node, sourceFile, this.newLine);
             if (this.validator) {
-                this.validator(nonFormattedText);
+                this.validator(nonformattedText);
             }
 
             const formatOptions = this.rulesProvider.getFormatOptions();
-            const pos = change.range.pos;
             const posStartsLine = getLineStartPositionForPosition(pos, sourceFile) === pos;
 
             const initialIndentation =
-                change.options.indentation !== undefined
-                    ? change.options.indentation
-                    : change.useIndentationFromFile
-                        ? formatting.SmartIndenter.getIndentation(change.range.pos, sourceFile, formatOptions, posStartsLine || (change.options.prefix === this.newLineCharacter))
+                options.indentation !== undefined
+                    ? options.indentation
+                    : (options.useIndentationFromFile !== false)

```

Comment:
We want `true` and `undefined` to behave the same here, hence the odd conditional